compiler/ast/types.nim

```diff
@@ -513,27 +499,52 @@ proc lastOrd*(conf: ConfigRef; t: PType): Int128 =
   of tyUInt64:
     result = toInt128(0xFFFFFFFFFFFFFFFF'u64)
   of tyEnum:
-    if t.n.len > 0:
-      assert(t.n[^1].kind == nkSym)
-      result = toInt128(t.n[^1].sym.position)
+    result =
+      if t.n.len > 0:
+        assert(t.n[^1].kind == nkSym)
+        toInt128(t.n[^1].sym.position)
+      else:
+        # currently empty enums aren't support as a type, but they should be
+        Zero
   of tyGenericInst, tyDistinct, tyTypeDesc, tyAlias, tySink,
      tyStatic, tyInferred, tyUserTypeClasses, tyLent:
     result = lastOrd(conf, lastSon(t))
-  of tyProxy: result = Zero
+  of tyProxy: result = Zero # xxx: this seems off; also not in `firstOrd`
```

Comment:
Given that `tyProxy` is only used for errors now, I think this should be handled as part of the `else` branch (i.e. treated as unreachable).

Adjusting it doesn't have to happen as part of this PR, though.

Reply:
I was thinking `tyProxy` branches are where we might add in forgiveness for suggest situations, while everything else is a programming error.

Reply:
Captured the above via comment.